### PR TITLE
Added new unit tests, cleaned up older ones

### PR DESCRIPTION
**Pull request by pmaytak**
Added unit tests for
- AzureADB2COpenIDConnectEventHandlersTests
- CookiePolicyOptionsExtensionsTests

Cleaned up some tests by using InlineData and changing some local variables to instance.

### CODE REVIEW

File: tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Identity.Web.Test
+{
+    public class AzureADB2COpenIDConnectEventHandlersTests
+    {
+        private const string PathBase = "/PathBase";
+        private const string DefaultUserFlow = "DefaultUserFlow";
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs]: >DefaultUserFlow [](start = 48, length = 15)

could you use something like "b2c_1_susi", which is closer to what the value would be? makes the test easier to understand for someone who doesn't know b2c

File: tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        private const string NonDefaultUserFlow = "NonDefaultUserFlow";
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs]: maybe call this "CustomUserFlow", which would be the non-default, and the value similar to something above, at least which starts w/"b2c_1_*"

File: tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        private string _defaultIssuer = $"IssuerAddress/{DefaultUserFlow}/";
+        private string _nonDefaultIssuer = $"IssuerAddress/{NonDefaultUserFlow}/";
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs]: not sure what you mean by default and non-default issuer and how that's impacted by the user flow? the user flow is independent of the issuer. maybe we can sync later, i think confused. :)

File: tests/Microsoft.Identity.Web.Test/Base64UrlHelpersTests.cs [hunk 4]
```diff
@@ -188,48 +88,18 @@ public void CreateString_NonUTF8Bytes_ReturnsInvalidString()
...
-            actualDecodedByteArray = Base64UrlHelpers.DecodeToBytes(stringToDecodeWithBase64Slash);
-
-            Assert.Equal(expectedDecodedByteArray, actualDecodedByteArray);
-
-            var emptyStringToDecode = string.Empty;
-            expectedDecodedByteArray = Encoding.UTF8.GetBytes(emptyStringToDecode);
+            var expectedDecodedByteArray = Encoding.UTF8.GetBytes(expectedDecodedString);
 
-            actualDecodedByteArray = Base64UrlHelpers.DecodeToBytes(emptyStringToDecode);
+            var actualDecodedByteArray = Base64UrlHelpers.DecodeToBytes(stringToDecode);
```
Comment (jennyf19) [inline tests/Microsoft.Identity.Web.Test/Base64UrlHelpersTests.cs]: yay to removing code!

File: tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs (hunk 3 above)
Comment (pmaytak) [reply tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs]: If the user flow in the `RedirectContext` is different from the default user flow (which is susi), then `IssuerAddress` is updated to replace the default user flow with the one from context.

https://github.com/AzureAD/microsoft-identity-web/blob/41798cebfe28a75f2984beca8407018fc9e77b72/src/Microsoft.Identity.Web/AzureADB2COpenIDConnectEventHandlers.cs#L28-L53

File: tests/Microsoft.Identity.Web.Test/Base64UrlHelpersTests.cs (hunk 4 above)
Comment (pmaytak) [reply tests/Microsoft.Identity.Web.Test/Base64UrlHelpersTests.cs]: 👍

File: tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs (hunk 1 above)
Comment (pmaytak) [reply tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs]: Agree. While the code under test doesn't specifically look for these values, updated to make it easier to understand.

File: tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs (hunk 2 above)
Comment (pmaytak) [reply tests/Microsoft.Identity.Web.Test/AzureADB2COpenIDConnectEventHandlersTests.cs]: Updated.

**Pull request merged**